### PR TITLE
fixing Databricks detector

### PR DESCRIPTION
**Pull request by 4390c336**
I'm just sending our internal developed detector

### Description:
we used this detector at Databricks for couple of months, 
and I noticed that the current detector is broken in many ways , so I just copy/pasted a working code.

image

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

**Comment by 4390c336** [comment]
the CLA check is not working. I singed the doc multiple times

**Comment by 4390c336** [comment]
@ahrav I didn't submit the PR for the competition, 
I just want this detector to be working so I don't have to build the code from source.

File: pkg/detectors/databrickstoken/databrickstoken.go [hunk 1]
```diff
@@ -21,52 +21,54 @@ var (
 	client = common.SaneHttpClient()
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	domain = regexp.MustCompile(`\b(https:\/\/[a-z0-9-]+\.cloud\.databricks\.com)\b`)
-	keyPat = regexp.MustCompile(`\b(dapi[a-z0-9]{32})\b`)
+	keyPat    = regexp.MustCompile(`\b(dapi[0-9Aa-f]{32})\b`)
```
Comment (rosecodym) [inline pkg/detectors/databrickstoken/databrickstoken.go:24]: Are you sure this pattern is correct? `[0-9Aa-f]` is an unusual character set. I would expect either `[0-9A-Fa-f]` or `[0-9a-f]`.

File: pkg/detectors/databrickstoken/databrickstoken.go [hunk 2]
```diff
@@ -21,52 +21,54 @@ var (
...
 
-// FromData will find and optionally verify Databrickstoken secrets in a given set of bytes.
+// FromData will find and optionally verify DatabricksToken secrets in a given set of bytes.
 func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
-	dataStr := string(data)
 
-	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
-	domainMatches := domain.FindAllStringSubmatch(dataStr, -1)
+	matches := keyPat.FindAll(data, -1)
+	domainMatches := domainPat.FindAll(data, -1)
```
Comment (rosecodym) [inline pkg/detectors/databrickstoken/databrickstoken.go:38]: If you use `FindAllString` instead of `FindAll` you won't have to manually convert back to `string` below.

File: pkg/detectors/databrickstoken/databrickstoken.go [hunk 3]
```diff
@@ -21,52 +21,54 @@ var (
...
+		resMatch := strings.TrimSpace(string(match))
+
+		for _, domainMatch := range domainMatches {
+
+			domain := strings.TrimSpace(string(domainMatch))
 
 			s1 := detectors.Result{
 				DetectorType: detectorspb.DetectorType_DatabricksToken,
 				Raw:          []byte(resMatch),
-				RawV2:        []byte(resMatch + resDomainMatch),
```
Comment (rosecodym) [inline pkg/detectors/databrickstoken/databrickstoken.go:56]: Why did you remove `RawV2`?

**Review by rosecodym — COMMENTED** [review]
Thank you for taking a look at this detector! We'd love to get some fixes in but I have some questions about the specifics.

File: pkg/detectors/databrickstoken/databrickstoken.go (hunk 1 above)
Comment (4390c336) [reply pkg/detectors/databrickstoken/databrickstoken.go:24]: it should be `[0-9a-f]` that was a mistake, thank you for spotting that. I will fix it right away

File: pkg/detectors/databrickstoken/databrickstoken.go (hunk 3 above)
Comment (4390c336) [reply pkg/detectors/databrickstoken/databrickstoken.go:56]: I wrote this detector long-time ago , not sure if RawV2 was there. I will add it. and test the code again

**Comment by 4390c336** [comment]
I updated the code based on the suggestions.
I also fixed a missing case with the regex where the token sometimes have an extra -\d at the end.

and I also submitted my _test.go since the existing one didn't expect a domain

File: pkg/detectors/databrickstoken/databrickstoken.go [hunk 4]
```diff
@@ -21,64 +20,66 @@ var (
 	client = common.SaneHttpClient()
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	domain = regexp.MustCompile(`\b(https:\/\/[a-z0-9-]+\.cloud\.databricks\.com)\b`)
-	keyPat = regexp.MustCompile(`\b(dapi[a-z0-9]{32})\b`)
+	keyPat    = regexp.MustCompile(`\b(dapi[0-9a-f]{32})(-\d)?\b`)
```
Comment (zricethezav) [inline pkg/detectors/databrickstoken/databrickstoken.go:23]: @4390c336 we're gonna prioritize merging https://github.com/trufflesecurity/trufflehog/pull/1965/files since it was opened first. The most important change is expanding the `domain` which both PRs include. However, I'm curious about this change to `keyPat`. Could you explain the reason behind this change? Do all databricks tokens have an appened `-{digit}`?

cc @ankushgoel27 (you don't need to do anything atm besides respond to the changes I requested in your PR, but wanted to make you aware of this one too)

Comment (4390c336) [reply pkg/detectors/databrickstoken/databrickstoken.go:23]: @zricethezav as indicated by regex . the `-{digit}` part is optional. not all tokens have it. some of them do.

source : I work for Databricks, and we have this running to scan our internal/public repos.

Comment (4390c336) [reply pkg/detectors/databrickstoken/databrickstoken.go:23]: PS : i don't care about the competition or credit as long as we merge a working detector

Comment (ankushgoel27) [reply pkg/detectors/databrickstoken/databrickstoken.go:23]: @zricethezav let me know if you would like to add the regex with optional digit added to my PR.

Comment (zricethezav) [reply pkg/detectors/databrickstoken/databrickstoken.go:23]: @ankushgoel27, yes please. I'll defer to @4390c336 as he is an engineer at Databricks.

**Comment by zricethezav** [comment]
Closing this in favor of @ankushgoel27's PR
